src/openqaoa-core/openqaoa/problems/bpsp.py

```diff
@@ -0,0 +1,617 @@
+
+        Attributes
+        ----------
+        car_sequence : list
+            A list representing the sequence of cars as they pass the paint shop.
+            Each car is represented by its identifier (e.g., integer), and appears twice in the sequence.
+        
+        car_positions : dict
+            A dictionary mapping each car to its two positions in the car_sequence.
+            For example, if car 'a' appears in positions 2 and 5 in car_sequence, then
+            car_positions['a'] = [2, 5].
+
+        Returns
+        -------
+        Model
+            A CPLEX model instance representing the BPSP, ready to be solved.
+        """
+        
+        mdl = Model("BPSP_Problem")
+        sequence = self.car_sequence
+        car_pos = self.car_positions
+
+        # Dictionary for storing the paint value for car 'x' at position 'j'.
+        w_vars = {f"{w}_{i}": mdl.binary_var(name=f"w_{w}_{i}") for i, w in enumerate(sequence)}
+
+        # This loop adds the constraint that a particular car cannot have the same paint
+        # at both occurrences in the paint sequence. If the first occurrence is 0, the other has to 
+        # be 1 and vice-versa.
+        for car, positions in car_pos.items():
+            w_key1, w_key2 = f"{car}_{positions[0]}", f"{car}_{positions[1]}"
+            mdl.add_constraint(w_vars[w_key1] + w_vars[w_key2] == 1)
+
+        # Encode the objective function: sum_i^2n-1 |paint[i] - paint[i+1]|. Since docplex accepts abs operator, 
+        # you can directly utilize it for the objective. This makes the model simpler than in Gurobi.
+        w_keys = list(w_vars.keys())
+        objective_function = mdl.sum(mdl.abs(w_vars[w_keys[i]] - w_vars[w_keys[i + 1]]) for i in range(len(w_keys) - 1))
+        
+        # Set the objective to minimize.
+        mdl.minimize(objective_function)
+
+        return mdl
+
+    @property
+    def qubo(self):
+        """
+        Returns the QUBO encoding of the Binary Paint Shop Problem.
+
+        This function provides a QUBO representation of the BPSP, where the objective is to minimize the
+        total weight of violations. The violations here refer to two adjacent cars in the sequence receiving 
+        the same color.
+
+        Returns
+        -------
+        QUBO
+            The QUBO encoding of the BPSP. This is a representation that describes 
+            the quadratic objective function over binary variables.
+        """
+        
+        # Iterate over edges (with weight) and store accordingly
+        terms = []
+        weights = []
+
+        # Here, we use self.bpsp_graph to gather terms and weights.
+        for u, v, edge_weight in self.bpsp_graph.edges(data="weight"):
+            terms.append([u, v])
+
+            weights.append(edge_weight)  
+
+        return QUBO(self.bpsp_graph.number_of_nodes(), terms, weights, self.problem_instance,)
+
+    def cplex_solution(self):
```

Comment:
Change the name of the function so it reflects its action, e.g. `find_cplex_solution()` or `solve_cplex()`

Reply:
Changed the name of the function to `solve_cplex`.